library/test/src/formatters/terse.rs

```diff
@@ -47,8 +47,12 @@ impl<T: Write> TerseFormatter<T> {
         self.write_short_result(".", term::color::GREEN)
     }
 
-    pub fn write_failed(&mut self) -> io::Result<()> {
-        self.write_short_result("F", term::color::RED)
+    pub fn write_failed(&mut self, name: &str) -> io::Result<()> {
+        // Put failed tests on their own line and include the test name, so that it's faster
+        // to see which test failed without having to wait for them all to run.
+        self.write_plain(format!("\n{name} --- "))?;
+        self.write_pretty("F", term::color::RED)?;
+        self.write_progress()
```

Comment:
I think this might want to print progress *first*, padded to the right alignment, because it looks awkward to me to have "count/total" on the same line as a single failure. But this could be debated.

Maybe we should also spell out `FAILED` here like the non-terse case?

Reply:
> I think this might want to print progress _first_, padded to the right alignment, because it looks awkward to me to have "count/total" on the same line as a single failure. But this could be debated.

:+1: i agree this looks better.

i don't feel strongly about F vs FAILED; i kind of like that F is shorter, since this is the terse format, but also it's already getting its own line so it's not that much shorter relatively. i've changed it to FAILED.